### PR TITLE
Log metric lines rejected by Dynatrace

### PR DESCRIPTION
**Pull request by dyladan**
During testing it was discovered that the Dynatrace metric exporter was logging errors incorrectly. This updates it to properly log the error message, and log lines which are rejected by the server at a debug level.

### CODE REVIEW

**Comment by dyladan** [comment]
@bogdandrutu looks like test failures are unrelated

File: exporter/dynatraceexporter/metrics_exporter.go [hunk 1]
```diff
@@ -164,12 +166,16 @@ func (e *exporter) send(ctx context.Context, message string) (int, error) {
...
-		if responseBody.Error != "" {
-			e.logger.Error(fmt.Sprintf("Error from Dynatrace: %s", responseBody.Error))
+		if responseBody.Error.Message != "" {
+			e.logger.Error(fmt.Sprintf("Error from Dynatrace: %s", responseBody.Error.Message))
 		}
 
-		return responseBody.Invalid, nil
+		for _, line := range responseBody.Error.InvalidLines {
+			// Enabled debug logging to see which lines were dropped
+			e.logger.Debug(fmt.Sprintf("line %3d %s %s", line.Line, line.Error, lines[line.Line]))
```
Comment (arminru) [inline exporter/dynatraceexporter/metrics_exporter.go:175]:
```suggestion
			e.logger.Debug(fmt.Sprintf("- rejected line %3d: %s [%s]", line.Line, line.Error, lines[line.Line]))
```

Comment (arminru) [inline exporter/dynatraceexporter/metrics_exporter.go:175]: `line.Line` as returned by the server should not be out of bounds but we should still check it nevertheless

**Review by arminru — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
@dyladan please fix the lint errors

**Pull request merged**